ballerina/types.bal

```diff
@@ -17,154 +17,396 @@
+    @display {label: "Auth Mechanism"}
+    readonly AUTH_GSSAPI authMechanism = AUTH_GSSAPI;
+    # The username for the database connection
     @display {label: "Username"}
     string username;
+    # The service name for the database connection. Use this to override the default service name of `mongodb`
     @display {label: "Service Name"}
     string serviceName?;
 |};
 
-# Represents the MongoDB connection pool properties
-# 
-# + readConcern - The read concern to use
-# + writeConcern - The write concern to use. The default value is `WriteConcern.ACKNOWLEDGED`
-# + readPreference - The read preference for the replica set
-# + authSource - The source in which the user is defined
-# + authMechanism - Authentication mechanism to use. 
-#                   Possible values are PLAIN, SCRAM_SHA_1, SCRAM_SHA_256, MONGODB-X509, or GSSAPI
-# + replicaSet - The replica set name if it is to connect to replicas
-# + sslEnabled - Whether SSL connection is enabled
-# + sslInvalidHostNameAllowed - Whether invalid host names should be allowed
-# + secureSocket - Configurations related to facilitating secure connection
-# + retryWrites - Whether to retry writing failures
-# + maxPoolSize - Maximum connection pool size
-# + minPoolSize - Minimum connection pool size
-# + socketTimeout - The socket timeout in milliseconds
-# + connectionTimeout - The connection timeout in milliseconds
-# + serverSelectionTimeout - The server selection timeout in milliseconds
-# + maxIdleTime - The maximum idle time for a pooled connection in milliseconds
-# + maxLifeTime - The maximum life time for a pooled connection in milliseconds
-# + waitQueueMultiple - The multiplier for the number of threads allowed to block waiting for a connection
-# + waitQueueTimeout - The maximum time that a thread will block waiting for a connection in milliseconds
-# + localThreshold - The local threshold latency in milliseconds
-# + heartbeatFrequency - The heartbeat frequency (ms). This is the frequency that the driver will attempt to 
-#                        determine the current state of each server in the cluster.
+# Represents the MongoDB connection pool properties.
 @display {label: "Connection Properties"}
 public type ConnectionProperties record {|
+    # The read concern level to use
     @display {label: "Read Concern"}
-    string readConcern?;
+    ReadConcern readConcern?;
+    # The write concern level to use
     @display {label: "Write Concern"}
     string writeConcern?;
+    # The read preference for the replica set
     @display {label: "Read Preference"}
     string readPreference?;
-    @display {label: "Auth Source"}
-    string authSource?;
-    @display {label: "Auth Mechanism"}
-    string authMechanism?;
+    # The replica set name if it is to connect to replicas
     @display {label: "Replica Set"}
     string replicaSet?;
+    # Whether SSL connection is enabled
     @display {label: "SSL Enabled"}
-    boolean sslEnabled?;
+    boolean sslEnabled = false;
```

Comment:
Why are we using a separate configuration to check SSL enabled or not ? Can't we use the `SecureSocket` config to identify this ?

Reply:
This is because the SSL can be used with default (well-known) certificates, where the user does not need to provide the `SecureSocket` configurations. Just enabling the SSL would be enough. It is useful when they connect with the MongoDB atlas database.